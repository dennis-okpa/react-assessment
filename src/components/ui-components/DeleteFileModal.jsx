import React from 'react';
import {
  Panel,
  ActionPanel,
  YesBtn,
  NoBtn
} from '@components/styled-components/Task';

const DeleteFileModal = () => (
  <Panel>
    <div>
      <h3>Are you sure you want to delete all of your files?</h3>
      <span>This action cannot be undone.</span>
    </div>
    <ActionPanel>
      <YesBtn onClick={(e) => { e.stopPropagation(); console.log('working yes'); }}>Yes</YesBtn>
      <NoBtn onClick={(e) => { e.stopPropagation(); console.log('working no'); }}>No</NoBtn>
    </ActionPanel>
  </Panel>
);

export default DeleteFileModal;
