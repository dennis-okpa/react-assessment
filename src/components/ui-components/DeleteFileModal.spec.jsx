import React from 'react';
import { render, fireEvent, act } from '@testing-library/react';
import DeleteFileModal from './DeleteFileModal';

const mockLog = jest.spyOn(global.console, 'log').mockImplementation();

describe('The DeleteFileModal Component', () => {
  describe('when rendered', () => {
    let container;
    let getByText;

    beforeEach(() => {
      ({ container, getByText } = render(
        <DeleteFileModal />
      ));
    });

    it('should match snapshot', () => {
      expect(container).toMatchSnapshot();
    });

    it('should call console with the correct string when Yes is clicked', async () => {
      await act(async () => {
        fireEvent.click(getByText('Yes'));
      });

      expect(mockLog).toHaveBeenCalledWith('yes sir');
    });

    it('should call console with the correct string when No is clicked', async () => {
      await act(async () => {
        fireEvent.click(getByText('No'));
      });

      expect(mockLog).toHaveBeenCalledWith('no sir');
    });
  });
});
