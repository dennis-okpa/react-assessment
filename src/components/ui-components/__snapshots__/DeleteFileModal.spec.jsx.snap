// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The DeleteFileModal Component when rendered should match snapshot 1`] = `
<div>
  <div
    class="Task__Panel-sc-7z955n-6 diyQSp"
  >
    <div>
      <h3>
        Are you sure you want to delete all of your files?
      </h3>
      <hr />
      <span>
        This action cannot be undone.
      </span>
    </div>
    <div
      class="Task__ActionPanel-sc-7z955n-9 jDSgIL"
    >
      <button
        class="Task__ActionBtn-sc-7z955n-10 Task__YesBtn-sc-7z955n-11 iozTJj bgrwqn"
      >
        Yes
      </button>
      <button
        class="Task__ActionBtn-sc-7z955n-10 Task__NoBtn-sc-7z955n-12 iozTJj dxGCKi"
      >
        No
      </button>
    </div>
  </div>
</div>
`;
